### ISSUE TITLE
Failed to enumerate view configuration views! [XR_ERROR_VALIDATION_FAILURE]

### ISSUE THREAD

**Issue by nocxtious** [GodotVR/godot_openxr#14]
Original issue: https://github.com/GodotVR/godot_openxr/issues/13

Using the same setup as posted in the above issue, changing the registry key to specify the Oculus OpenXR runtime results in the following log output/error:
```Godot Engine v3.2.4.beta.custom_build.61c8efecf - https://godotengine.org
OpenGL ES 3.0 Renderer: AMD Radeon RX 5700 XT
OpenGL ES Batching: ON

Editing project: F:/Godot/Projects/3_2_branch_openxr_test_01 (F:::Godot::Projects::3_2_branch_openxr_test_01)
Godot Engine v3.2.4.beta.custom_build.61c8efecf - https://godotengine.org
OpenGL ES 3.0 Renderer: AMD Radeon RX 5700 XT
OpenGL ES Batching: ON

Running: F:\Godot\Branches\3.2\godot\bin\godot.windows.tools.64.exe --path F:/Godot/Projects/3_2_branch_openxr_test_01 --remote-debug 127.0.0.1:6007 --allow_focus_steal_pid 20076 --position 448,240 res://ARVROrigin.tscn
Godot Engine v3.2.4.beta.custom_build.61c8efecf - https://godotengine.org
OpenGL ES 3.0 Renderer: AMD Radeon RX 5700 XT
OpenGL ES Batching: ON

OpenXR: initialising OpenXR context
Failed to enumerate view configuration views! [XR_ERROR_VALIDATION_FAILURE]
OpenXR init failed
```

Original (default) registry key:
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\OpenXR\1
ActiveRuntime: D:\Steam\steamapps\common\SteamVR\steamxr_win64.json

Modified key:
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\OpenXR\1
ActiveRuntime: D:\Oculus\Support\oculus-runtime\oculus_openxr_64.json

Windows 10 x64 20H2 19042.572
Radeon Driver: 20.10.1
Godot Verison: Godot Engine v3.2.4.beta.custom_build.61c8efecf
OpenXR Plugin: (last commit 7e5c2a9)
Using the provided OpenXR Loader .dll in repo (do I need to compile a new one for x64?)
Oculus Quest 1 using Oculus Link
Oculus App Version: 21.0.0.473.199
Quest Firmware: 904120.19760.0
SteamVR: 1.14.16

Please let me know if there is any additional information or steps you would like me to perform or provide that could be helpful/useful!

Relavent links:
https://forums.oculusvr.com/developer/discussion/comment/761189/#Comment_761189
https://steamcommunity.com/app/250820/discussions/8/2647504242046572684/
https://steamcommunity.com/app/250820/discussions/8/2647504242044701031/

**Comment by nocxtious** [comment]
Enabled validation layers and reran existing project again with both Oculus and SteamVR runtimes, output below.

Oculus:
```Running: F:\Godot\Branches\3.2\godot\bin\godot.windows.tools.64.exe --path F:/Godot/Projects/3.2/openxr_test_01 --remote-debug 127.0.0.1:6007 --allow_focus_steal_pid 9400 --position 448,240
Godot Engine v3.2.4.beta.custom_build.61c8efecf - https://godotengine.org
OpenGL ES 3.0 Renderer: AMD Radeon RX 5700 XT
OpenGL ES Batching: ON

OpenXR Runtime: D:\Oculus\Support\oculus-runtime\oculus_openxr_64.json
OpenXR: initialising OpenXR context
Core Validation output type: text, first time = true
[VALID_ERROR | VUID-XrViewConfigurationView-type-type | xrEnumerateViewConfigurationViews]: XrViewConfigurationView has an invalid XrStructureType 0x004f0053, expected 0x004f0053 (XR_TYPE_VIEW_CONFIGURATION_VIEW)
  Objects:
   [0] - XrInstance (0x0000213800000001)
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] std::_Find_unchecked1<enum XrStructureType *,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5133)
[1] std::_Find_unchecked<enum XrStructureType *,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5168)
[2] std::find<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<enum XrStructureType> > >,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5173)
[3] ValidateNextChain (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:11369)
[4] ValidateXrStruct (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:4391)
[5] GenValidUsageInputsXrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:13686)
[6] GenValidUsageXrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:13727)
[7] xrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK\src\loader\xr_generated_loader.cpp:282)
[8] godot_openxr_gdnative_init
[9] godot_openxr_gdnative_init
[10] godot_opexr_nativescript_init
[11] ARVRInterfaceGDNative::initialize (F:\Godot\Branches\3.2\godot\modules\gdnative\arvr\arvr_interface_gdnative.cpp:150)
[12] MethodBind0R<ARVRInterface,bool>::call (F:\Godot\Branches\3.2\godot\core\method_bind.gen.inc:325)
[13] Object::call (F:\Godot\Branches\3.2\godot\core\object.cpp:922)
[14] Variant::call_ptr (F:\Godot\Branches\3.2\godot\core\variant_call.cpp:1128)
[15] GDScriptFunction::call (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript_function.cpp:1086)
[16] GDScriptInstance::_ml_call_reversed (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript.cpp:1270)
[17] GDScriptInstance::call_multilevel_reversed (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript.cpp:1279)
[18] Node::_notification (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:152)
[19] Node::_notificationv (F:\Godot\Branches\3.2\godot\scene\main\node.h:46)
[20] Spatial::_notificationv (F:\Godot\Branches\3.2\godot\scene\3d\spatial.h:54)
[21] ARVROrigin::_notificationv (F:\Godot\Branches\3.2\godot\scene\3d\arvr_nodes.h:154)
[22] Object::notification (F:\Godot\Branches\3.2\godot\core\object.cpp:934)
[23] Node::_propagate_ready (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:197)
[24] Node::_propagate_ready (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:189)
[25] Node::_set_tree (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:2574)
[26] SceneTree::init (F:\Godot\Branches\3.2\godot\scene\main\scene_tree.cpp:464)
[27] OS_Windows::run (F:\Godot\Branches\3.2\godot\platform\windows\os_windows.cpp:3437)
[28] widechar_main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:162)
[29] _main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:184)
[30] main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:196)
[31] __scrt_common_main_seh (D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[32] BaseThreadInitThunk
-- END OF BACKTRACE --
```

SteamVR:
```OpenXR Runtime: D:\Steam\steamapps\common\SteamVR\steamxr_win64.json
OpenXR: initialising OpenXR context
Core Validation output type: text, first time = true
[VALID_ERROR | VUID-XrViewConfigurationView-type-type | xrEnumerateViewConfigurationViews]: XrViewConfigurationView has an invalid XrStructureType 0xdddddddd, expected 0xdddddddd (XR_TYPE_VIEW_CONFIGURATION_VIEW)
  Objects:
   [0] - XrInstance (0x000002e2366b60f0)
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] std::_Find_unchecked1<enum XrStructureType *,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5133)
[1] std::_Find_unchecked<enum XrStructureType *,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5168)
[2] std::find<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<enum XrStructureType> > >,enum XrStructureType> (F:\Program Files (x86)\VC\Tools\MSVC\14.27.29110\include\xutility:5173)
[3] ValidateNextChain (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:11369)
[4] ValidateXrStruct (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:4391)
[5] GenValidUsageInputsXrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:13686)
[6] GenValidUsageXrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK-Source\build\win64\src\api_layers\xr_generated_core_validation.cpp:13727)
[7] xrEnumerateViewConfigurationViews (F:\Godot\Plugins\OpenXR-SDK\src\loader\xr_generated_loader.cpp:282)
[8] godot_openxr_gdnative_init
[9] godot_openxr_gdnative_init
[10] godot_opexr_nativescript_init
[11] ARVRInterfaceGDNative::initialize (F:\Godot\Branches\3.2\godot\modules\gdnative\arvr\arvr_interface_gdnative.cpp:150)
[12] MethodBind0R<ARVRInterface,bool>::call (F:\Godot\Branches\3.2\godot\core\method_bind.gen.inc:325)
[13] Object::call (F:\Godot\Branches\3.2\godot\core\object.cpp:922)
[14] Variant::call_ptr (F:\Godot\Branches\3.2\godot\core\variant_call.cpp:1128)
[15] GDScriptFunction::call (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript_function.cpp:1086)
[16] GDScriptInstance::_ml_call_reversed (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript.cpp:1270)
[17] GDScriptInstance::call_multilevel_reversed (F:\Godot\Branches\3.2\godot\modules\gdscript\gdscript.cpp:1279)
[18] Node::_notification (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:152)
[19] Node::_notificationv (F:\Godot\Branches\3.2\godot\scene\main\node.h:46)
[20] Spatial::_notificationv (F:\Godot\Branches\3.2\godot\scene\3d\spatial.h:54)
[21] ARVROrigin::_notificationv (F:\Godot\Branches\3.2\godot\scene\3d\arvr_nodes.h:154)
[22] Object::notification (F:\Godot\Branches\3.2\godot\core\object.cpp:934)
[23] Node::_propagate_ready (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:197)
[24] Node::_propagate_ready (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:189)
[25] Node::_set_tree (F:\Godot\Branches\3.2\godot\scene\main\node.cpp:2574)
[26] SceneTree::init (F:\Godot\Branches\3.2\godot\scene\main\scene_tree.cpp:464)
[27] OS_Windows::run (F:\Godot\Branches\3.2\godot\platform\windows\os_windows.cpp:3437)
[28] widechar_main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:162)
[29] _main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:184)
[30] main (F:\Godot\Branches\3.2\godot\platform\windows\godot_windows.cpp:196)
[31] __scrt_common_main_seh (D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[32] BaseThreadInitThunk
-- END OF BACKTRACE --
```
Please let me know if there is anything else I can run or provide that might assist

**Comment by ChristophHaag** [comment]
Thanks for the log. For some reason we didn't notice this but I could actually reproduce it here. Also fixed another issue in  #16 when running with validation layer enabled.

I believe those were still some collateral damage from the C to C++ port of the plugin.

**Comment by nocxtious** [comment]
Can confirm building https://github.com/GodotVR/godot_openxr/pull/16 is working with the Oculus runtime and no longer throwing validation errors and showing video properly in headset.  SteamVR problems persist as described in https://github.com/GodotVR/godot_openxr/issues/13 but I think that is expected in its current state based off the forum links.
